CONTRIBUTING.txt

```diff
@@ -0,0 +1,99 @@
+    3. To preserve and improve the code quality while encouraging appropriate experimentation and risk-taking.
+    4. To minimize dependence on individual Contributors by encouraging a large pool of active Contributors.
+    5. To be inclusive of many viewpoints and to harness a diverse set of skills.
+    6. To provide an encouraging environment where Contributors learn and improve their skills.
+    7. To protect the free and open nature of the FreeCAD project.
+
+2. Fundamentals
+
+    1. FreeCAD uses the git distributed revision control system.
+    2. Source code for the main application and related subprojects is hosted on github.com in the FreeCAD organization, hereafter referred to as 'the Platform'.
+    3. Problems are discrete, well-defined limitations or bugs.
+    4. FreeCAD uses the Platform's issue-tracking system to track problems.
+    5. Patches are sets of code changes that resolve a single problem.
+    6. FreeCAD uses the Pull Request workflow for evaluating and accepting patches.
+
+3. Roles
+    1. "User": An member of the wider FreeCAD community who uses the software.
+    2. "Contributor":  A person who submits a patch that resolves a previously identified problem. Contributors do not have commit access to the repository unless they are also Maintainers. Everyone, without distinction or discrimination, SHALL have an equal right to become a Contributor.
+    3. "Maintainer": A person who merges patches. Maintainers may or may not be Contributors. Their role is to enforce the process. Maintainers have commit access to the repository.
+    4. "Administrator": Administrators have additional authority to maintain the list of designated Maintainers.
+
+4. Licensing, Ownership, and Credit
+    1. FreeCAD is distributed under the Lesser General Public License, version 2, or superior (LGPL2+).  Additional details can be found in the LICENSE file.
+    2. All contributions to FreeCAD MUST use a compatible license.
+    3. All contributions are owned by their authors unless assigned to another.
+    4. FreeCAD does not have a mandatory copyright assignment policy.
+    5. A Contributor who wishes to be identified in the Credits section of the application "About" dialog is responsible for identifying themselves appropriately by notifying an Administrator or Maintainer.
+
+5. Contribution Requirements
+
+    1. Patches are submitted in the form of Pull Requests (PR).
+    2. Maintainers and Contributors MUST have a Platform account and SHOULD use their real names or a well-known alias. If the Platform alias differs from the account alias on the FreeCAD Forum, effort SHOULD be taken to avoid confusion.
+    3. A PR SHOULD be a minimal and accurate answer to exactly one identified and agreed-on problem.
+    4. A PR SHOULD refrain from adding additional dependencies to the FreeCAD project unless no other option is available.
+    5. Code submissions MUST adhere to the code style guidelines of the project if these are defined.
```

Comment:
Doesn't that just make the situation more complicated?  Now you have to define what is a "small change".  
Let's keep it simple and add complexity later if needed.

Reply:
I think MUST is not the best word to use here, if a file is not following the defined style it is IMO better to keep the style it uses so that it is uniform instead of mixing. And of course we can't ask someone that just wants to fix something to reformat the entire file before merging

Reply:
Why can't we ask someone who is changing a file to reformat the whole file before changing it?  It usually takes one command and a few seconds for most programming languages.  Python has "black" and several other possible tools.  I believe that there are equivalent tools for C++.
This is also an issue that rapidly goes away as all "active" files get reformatted.

Reply:
Please continue discussion of code formatting policy on this issue  https://github.com/FreeCAD/FreeCAD/issues/7650